src/js/modules/utilities/Logging.ts

```diff
@@ -9,40 +9,99 @@
 
-export function newLogger(
-  module_name: string,
-  logLevel = "info",
-  simplify = false
-) {
-  const rpWasmFormatter = () => {
-    const logLineFormatter = winston.format.printf(
-      ({ level, message, label, timestamp }) => {
-        return `${timestamp} [${label}] ${level}: ${message}`;
-      }
-    );
-    const simpleLogLineFormatter = winston.format.printf(
-      ({ level, message, label }) => {
-        return `[${label}] ${level}: ${message}`;
-      }
-    );
+const maxLogFileSize = 5000000;
+
+const rpWasmFormatter = (moduleName: string, simplify: boolean) => {
+  const logLineFormatter = winston.format.printf(
+    ({ level, message, label, timestamp }) => {
+      return `${timestamp} [${label}] ${level}: ${message}`;
+    }
+  );
+  const simpleLogLineFormatter = winston.format.printf(
+    ({ level, message, label }) => {
+      return `[${label}] ${level}: ${message}`;
+    }
+  );
+
+  return (simplify ? [] : [winston.format.timestamp()]).concat([
+    winston.format.splat(),
+    winston.format.label({ label: moduleName }),
+    simplify ? simpleLogLineFormatter : logLineFormatter,
+  ]);
+};
+
+export class LogService {
+  private static instance: LogService;
+  private logFilePath: string;
+  private winstonContext: Container;
+
+  constructor() {
+    this.winstonContext = new winston.Container();
+  }
+
+  static getInstance(): LogService {
+    if (this.instance == undefined) {
+      this.instance = new LogService();
+      return this.instance;
+    }
+    return this.instance;
+  }
 
-    return (simplify ? [] : [winston.format.timestamp()]).concat([
-      winston.format.splat(),
-      winston.format.label({ label: module_name }),
-      simplify ? simpleLogLineFormatter : logLineFormatter,
-    ]);
-  };
-
-  const logger = winston.createLogger({
-    level: logLevel,
-    format: winston.format.combine(...rpWasmFormatter()),
-    defaultMeta: { service: module_name },
-    transports: [
-      new winston.transports.Console({
-        level: logLevel,
-      }),
-    ],
-  });
-  return logger;
+  setPath(path: string): void {
+    if (this.logFilePath == undefined) {
+      this.logFilePath = path;
+    }
+  }
+
+  getPath(): string {
+    return this.logFilePath;
+  }
+
+  createLogger(
+    moduleName: string,
+    option?: LoggerOptions,
+    simplify = false
+  ): Logger {
+    const loggerConfig = {
+      ...option,
+      level: option?.level,
+      format: winston.format.combine(...rpWasmFormatter(moduleName, simplify)),
+      defaultMeta: { service: moduleName },
+      transports: [
+        new winston.transports.Console({
+          level: option?.level,
+        }),
+        new winston.transports.File({
+          maxsize: maxLogFileSize,
+          filename: this.logFilePath,
+          level: option?.level,
+        }),
+      ],
+      exitOnError: false,
+    };
+    const prevModule = this.winstonContext.has(moduleName);
+    if (prevModule) {
```

Comment:
This if statement should come before you create a loggerConfig. Otherwise you would have created a loggerConfig you wouldn't have used in the case the 'if' statement is true and you just return an existing config

Reply:
yes, that's make sense

Reply:
Done